### PR TITLE
Web3 1.0 changes

### PR DESCRIPTION
**Pull request by justinkchen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,31 +69,34 @@
...
     "husky": "^0.14.3",
     "lint-staged": "^7.2.0",
     "module-alias": "^2.1.0",
-    "set-protocol-utils": "^0.3.37",
+    "set-protocol-utils": "^1.0.0-beta.10",
     "sol-trace-set": "^0.0.1",
     "solium": "^1.1.7",
     "tiny-promisify": "^1.0.0",
-    "truffle-hdwallet-provider": "^0.0.3",
+    "truffle-hdwallet-provider": "^1.0.0-web3one.0",
```
Comment (felix2feng) [inline package.json]: That's cool this dep has a web3 1.0 version

File: tsconfig.dist.json [hunk 2]
```diff
@@ -15,23 +15,25 @@
...
         "node_modules/@0xproject/types/*",
-        "node_modules/@0xproject/typescript-typings/types/*",
+        "node_modules/set-protocol-utils/types/*",
         "node_modules/*"
       ]
     },
     "declaration": true,
-    "declarationDir": "./dist/typings"
+    "declarationDir": "./dist/typings",
+    "esModuleInterop": true
```
Comment (felix2feng) [inline tsconfig.dist.json]: What's the benefit / need for `esModuleInterop`?

File: types/base_contract.ts [hunk 3]
```diff
@@ -14,14 +14,14 @@ export class BaseContract {
     public abi: any[];
     // public abi: Web3.AbiDefinition[];
 
-    public web3ContractInstance: Web3.ContractInstance;
+    public web3ContractInstance: any;
```
Comment (felix2feng) [inline types/base_contract.ts]: I get really nervous when we're moving to an `any` type. Is there another type we could use?

Comment (felix2feng) [reply types/base_contract.ts]: If it doesn't exist, we should consider creating our own type

File: tsconfig.dist.json (hunk 2 above)
Comment (asoong) [reply tsconfig.dist.json]: https://www.typescriptlang.org/docs/handbook/compiler-options.html

File: truffle.js [hunk 4]
```diff
@@ -38,7 +38,15 @@ module.exports = {
       gas: 0xfffffffffff,
       gasPrice: 0x01,
     },
-  }
+  },
+  compilers: {
```
Comment (asoong) [inline truffle.js]: do we still need the top level `solc:` option?

File: package.json (hunk 1 above)
Comment (justinkchen) [reply package.json]: it is required as part of the truffle 5 migration efforts too i think

File: tsconfig.dist.json (hunk 2 above)
Comment (justinkchen) [reply tsconfig.dist.json]: Required in order to allow for default imports (so we don't have to do `require('...')`) for modules that don't explicitly have that

File: types/base_contract.ts (hunk 3 above)
Comment (justinkchen) [reply types/base_contract.ts]: There is no other type right now defined at least.  `web3.eth.Contract` is a class that has a constructor to generate classes, but I couldn't get it working as a type yet

File: truffle.js (hunk 4 above)
Comment (justinkchen) [reply truffle.js]: i think we may be able to get rid of that

File: types/base_contract.ts (hunk 3 above)
Comment (justinkchen) [reply types/base_contract.ts]: i guess same thing should be done for the `// public abi: Web3.AbiDefinition[];` comment we have above. not sure why it was commented to be `any` there as well

File: truffle.js (hunk 4 above)
Comment (justinkchen) [reply truffle.js]: interesting, i know it picks up the compiler version, but maybe the solc is not somehow...or maybe its in your local truffle version not using it properly? Didn't you say you weren't on truffle 5 yet?

Comment (justinkchen) [reply truffle.js]: we can also keep for backwards compatibility on older truffle versions

File: tsconfig.dist.json (hunk 2 above)
Comment (justinkchen) [reply tsconfig.dist.json]: in particular i think web3 and some other package had this problem that this was able to solve

**Pull request merged**